packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -542,7 +541,7 @@ class _DragTargetState<T> extends State<DragTarget<T>> {
       _rejectedAvatars.remove(avatar);
     });
     if (widget.onLeave != null)
-      widget.onLeave(avatar.data);
+      widget.onLeave(avatar.data as T);
```

Comment:
This one is strange because IIUC avatar could be in `_rejectedAvatars` (see the above assert) and so it could be something that is not a `T`.
Unfortunately I don't have enough context to know if it is really a problem and how to fix it.

Reply:
Should this just be `as dynamic` until we determine if the type annotations are correct?

Reply:
Hm.  Looks like that won't work.  I strongly suspect we just want the `_rejectedAvatars` collection to be `<T>` instead of `<dynamic>`.

Reply:
I initially tried to make some changes (like retype `_rejectedAvatars`) but there was always a test that failed.

Reply:
_rejectedAvatatars might not be T, it could be anything. Imagine you have two drag sources, one with Xs and one with Ys, and two drag targets, one for Xs and one for Ys. Drag from the X source to the Y target and it'll be an X rejected in the Y.

Reply:
looks like onLeave is just badly defined, should be types as `<Object>` rather than `T`.

BTW, avoid `dynamic` as much as possible. Prefer `Object`.

Reply:
Done